File: ejercicios/c++/ejercicio058.cpp
#include <iostream>
using namespace std;

long cuenta_digitos_iterativo(int numero){	int contador = 0;
	if(numero < 0)
		numero *= -1;
	if(numero == 0)
		contador = 1;
	while(numero > 0){
		contador++;
		numero /= 10;
	}
	return contador;
}

int main() {
	int numero;
	cout << "ingrese el numero : ";
	cin >> numero;
	
	cout << numero << " tiene : " << cuenta_digitos_iterativo(numero) << " digitos " << endl;
	
	return 0;
}

File: ejercicios/python/ejercicio058.py
def cuenta_digitos_iterativo(numero):
   contador = 0
   if numero < 0:
      numero *= -1
   if numero == 0:
      contador = 1
   while numero > 0:
      contador += 1
      numero = int(numero / 10)
   return contador

if __name__ == "__main__":
   numero = int(input("ingrese el numero : "))
   print (str(numero) + " tiene : " + str(cuenta_digitos_iterativo(numero)) + " digitos " )
